### PR TITLE
refactor update_jobs_statuses

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
fix: #907 

### Details and comments

put some logic in separate functions

### CODE REVIEW

File: gateway/api/schedule.py [hunk 1]
```diff
@@ -153,3 +155,41 @@ def get_jobs_to_schedule_fair_share(slots: int) -> List[Job]:
         job_filter |= Q(author=entry["author"]) & Q(created=entry["job_date"])
 
     return Job.objects.filter(job_filter)
+
+
+def check_job_timeout(job: Job, job_status):
```
Comment (psschwei) [inline gateway/api/schedule.py:160]: I always get a bit mixed up on how Python handles passing variables, so just want to ask as a sanity check that the updates made to the `job` within these functions are seen outside of these functions, right?

Comment (akihikokuroda) [reply gateway/api/schedule.py:160]: That's correct.  I looked it up :-) and also tested.

**Review by psschwei — APPROVED** [review]

File: gateway/api/schedule.py [hunk 2]
```diff
@@ -153,3 +155,41 @@ def get_jobs_to_schedule_fair_share(slots: int) -> List[Job]:
...
+    if job.updated and endtime < now:
+        job_status = Job.STOPPED
+        job.logs = f"{job.logs}.\nMaximum job runtime reached. Stopping the job."
+        job.save()
+        logger.warning(
+            "Job [%s] reached maximum runtime [%s] days and stopped.",
+            job.id,
+            timeout,
+        )
+    print(job_status)
```
Comment (IceKhan13) [inline gateway/api/schedule.py:176]: Remove this and we are good :)

**Review by IceKhan13 — APPROVED** [review]
Great! much cleaner!

**Pull request merged**